### PR TITLE
Shorten Curve arrowstyle implementations.

### PR DESCRIPTION
**Pull request by anntzer**
... by moving {begin,end}arrow and fill{begin,end} to class attributes.

## PR Checklist

- [ ] Has pytest style unit tests (and `pytest` passes).
- [ ] Is [Flake 8](https://flake8.pycqa.org/en/latest/) compliant (run `flake8` on changed files to check).
- [ ] New features are documented, with examples if plot related.
- [ ] Documentation is sphinx and numpydoc compliant (the docs should [build](https://matplotlib.org/devel/documenting_mpl.html#building-the-docs) without error).
- [ ] Conforms to Matplotlib style conventions (install `flake8-docstrings` and run `flake8 --docstring-convention=all`).
- [ ] New features have an entry in `doc/users/next_whats_new/` (follow instructions in README.rst there).
- [ ] API changes documented in `doc/api/next_api_changes/` (follow instructions in README.rst there).

### CODE REVIEW

File: lib/matplotlib/patches.py [hunk 1]
```diff
@@ -2968,6 +2968,13 @@ def __call__(self, path, mutation_size, linewidth,
             else:
                 return self.transmute(path, mutation_size, linewidth)
 
+    @_register_style(_style_list, name="-")
+    class Curve(_Base):
```
Comment (QuLogic) [inline lib/matplotlib/patches.py:2972]: Why the superclass change? Now it won't have the `beginarrow` properties.

Comment (anntzer) [reply lib/matplotlib/patches.py:2972]: The original idea was that none of these attributes make sense for Curve anyways, and that this way I don't have to hide head_length/head_width by writing a separate `__init__` (I could have restored beginarrow/etc. as deprecated (or not deprecated) attributes), but OTOH I guess someone *could* have been doing `blah.get_arrowstyle().beginarrow = True` to switch between Curve and Arrow, so I decided to go back to the old design.

File: lib/matplotlib/patches.py [hunk 2]
```diff
@@ -3102,113 +3102,43 @@ def transmute(self, path, mutation_size, linewidth):
     class Curve(_Curve):
         """A simple curve without any arrow head."""
 
-        def __init__(self):
-            super().__init__(beginarrow=False, endarrow=False)
+        # These attributes only matter if someone modifies beginarrow/etc. on
+        # an ArrowStyle instance.
+        head_length = .2
+        head_width = .1
```
Comment (QuLogic) [inline lib/matplotlib/patches.py:3108]: This doesn't get overwritten by the super `__init__`, which has different defaults?

File: doc/api/next_api_changes/deprecations/20105-AL.rst [hunk 3]
```diff
@@ -0,0 +1,3 @@
+ArrowStyle.Curve attributes
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
+The ``
```
Comment (QuLogic) [inline doc/api/next_api_changes/deprecations/20105-AL.rst:3]: Incomplete...

File: lib/matplotlib/patches.py (hunk 2 above)
Comment (anntzer) [reply lib/matplotlib/patches.py:3108]: I think that's correct for backcompat?  Previously, `Curve.__init__` would call `_Curve.__init__` and use the default head_length and head_width from it, which were .2, .1 (and not .4, .2 as they are for all other `_Curve` subclasses), so I just hard-coded these values here?

File: doc/api/next_api_changes/deprecations/20105-AL.rst (hunk 3 above)
Comment (anntzer) [reply doc/api/next_api_changes/deprecations/20105-AL.rst]: oops, removed (that is a remnant of when I was thinking of removing the attributes on Curve.

File: lib/matplotlib/patches.py (hunk 2 above)
Comment (QuLogic) [reply lib/matplotlib/patches.py:3108]: But the `Curve.__init__` call to `super().__init__()` passes nothing, so the defaults from the signature will be used, which will overwrite these two.

Comment (anntzer) [reply lib/matplotlib/patches.py]: ah, indeed; moved them to instance attributes instead.

Comment (QuLogic) [reply lib/matplotlib/patches.py]: Or you could pass the different default to `super().__init__()`.

**Review by QuLogic — APPROVED** [review]

File: lib/matplotlib/patches.py (hunk 2 above)
Comment (anntzer) [reply lib/matplotlib/patches.py]: Indeed, I guess that's a bit nicer.

File: lib/matplotlib/patches.py [hunk 4]
```diff
@@ -2971,24 +2971,24 @@ def __call__(self, path, mutation_size, linewidth,
...
+        returned path is the concatenation of the original path, and at
         most two paths representing the arrow head at the begin point and the
         at the end point. The arrow heads can be either open or closed.
         """
 
-        def __init__(self, beginarrow=None, endarrow=None,
-                     fillbegin=False, fillend=False,
-                     head_length=.2, head_width=.1):
+        beginarrow = endarrow = False  # Whether arrows are drawn.
+        fillbegin = fillend = False  # Whether arrows are filled.
```
Comment (jklymak) [inline lib/matplotlib/patches.py:2980]: codecov seems pretty broken these days, but *is* this actually tested?

**Review by jklymak — APPROVED** [review]

**Pull request merged**